Sword

Startarrot opened the pull request:
Mostly done.  kill isn't working right

Startarrot committed:
useItem v1

Startarrot committed:
SwordSwing

SamBumgardner commented on source/actors/player/Hero.hx:
```diff
@@ -39,6 +39,15 @@ class Hero extends FlxSprite {
         standardMovement(moveDirection);
     }
 
+    public function getX()
+        {
+            return x;
+        }
+    public function getY()
+        {
+            return y;
+        }
+
```
External objects should just be able to access the `x` and `y` properties of the sword directly. Was there a particular reason for using these getter methods?

SamBumgardner commented on source/actors/player/Item.hx:
```diff
@@ -0,0 +1,80 @@
+        super();
+    }
+    override public function update(elapsed:Float):Void
+        {
+            if(FlxG.keys.pressed.RIGHT)
+                {
+                    lastUsed = 0;
+                }
+            else if(FlxG.keys.pressed.DOWN)
+                {
+                    lastUsed = 1;
+                }
+            else if(FlxG.keys.pressed.LEFT)
+                {
+                    lastUsed = 2;
+                }
+            else if(FlxG.keys.pressed.UP)
+                {
+                    lastUsed = 3;
+                }
```
the logic here can get sketchy if the user presses multiple keys at the same time (if-else if-else block is all exclusive, so there's some built in priority going on that doesn't necessarily give us the result we want).

Instead of making the item responsible for keeping track of this "facing" information, I recommend using the hero's `facing` property (defined in the FlxSprite class: https://api.haxeflixel.com/flixel/FlxSprite.html#facing). That variable isn't updated automatically by anything in the game right now, but you can add logic to the movement direction-selecting function to make it set `this.facing = FlxObject.LEFT` or other directions as needed.

SamBumgardner commented on source/actors/player/Item.hx:
```diff
@@ -0,0 +1,80 @@
+                {
+                    lastUsed = 3;
+                }
+            if(time.finished)
+            {
+                time.cancel();
+                //kill();
+            }
```
I don't think this `cancel` call should be needed, and it might be messing with the timer's execution of the `OnComplete` function that you set up? (`stopSwing`, in this case)

I'd need to do some direct testing to know for sure.

SamBumgardner commented on source/actors/player/Item.hx:
```diff
@@ -0,0 +1,80 @@
+                angle = 180;
+                makeGraphic(10, 50);
+            }
+        time = new FlxTimer().start(delay, stopSwing, 1);
```
Let's create one FlxTimer object in the constructor and just reuse it here instead of creating a new FlxTimer object each time `useItem` is called.

Startarrot committed:
Sword Despawn

Despawning of sword works now.  Facing is a issue

Startarrot committed:
minor tweak

Startarrot committed:
Sword update

Almost done.  little stutter can come up when swinging.

SamBumgardner committed:
Cleaned up some sword logic in-class.

Startarrot committed:
Merge pull request #1 from SamBumgardner/sword-fixes

Cleaned up some sword logic in-class.

Startarrot committed:
Fixed Swords

Fixed swords.  They strike in the four directions and should have proper collision boxes.

Startarrot committed:
Merge branch 'sword' of https://github.com/Startarrot/csc-303-game-2021 into sword

Startarrot committed:
Merge branch 'main' into sword

Startarrot committed:
resized swords

swords are resized for the smaller hero.

SamBumgardner commented on source/actors/player/Item.hx:
```diff
@@ -0,0 +1,75 @@
+    
+    public function new(owner:FlxSprite) {
+        super();
+        makeGraphic(0,0);
```
A graphic with dimensions 0,0 causes an exception in the code (the game doesn't crash, but it shows up if you run it in debug mode with `lime test neko -debug`
```suggestion
        makeGraphic(1,1);
```

Startarrot committed:
Update source/actors/player/Item.hx

Co-authored-by: Sam Bumgardner <Sambumgardner1@gmail.com>
